.github/workflows/test.yaml

```
@@ -6,9 +6,12 @@ on: [push]
 jobs:
   test:
     runs-on: ubuntu-latest
-    container: python:3.9
+    container: python:3.11
 
     steps:
+      - name: Delete huge unnecessary tools folder
+        run: rm -rf /opt/hostedtoolcache
+
```

I don't _think_ you need this, at least for running the tests

```suggestion
      - name: Delete huge unnecessary tools folder
        run: rm -rf /opt/hostedtoolcache

```